optaplanner-core/src/main/java/org/optaplanner/core/impl/heuristic/selector/move/generic/list/ListAssignMove.java

```diff
@@ -68,33 +93,19 @@ public String getSimpleMoveTypeDescription() {
         return getClass().getSimpleName() + "(" + variableDescriptor.getSimpleEntityAndVariableName() + ")";
     }
 
-    // ************************************************************************
-    // Testing methods
-    // ************************************************************************
-
-    public Object getDestinationEntity() {
-        return destinationEntity;
-    }
-
-    public int getDestinationIndex() {
-        return destinationIndex;
-    }
-
-    public Object getMovedValue() {
-        return planningValue;
-    }
-
     @Override
     public boolean equals(Object o) {
         if (this == o) {
             return true;
         }
-        if (!(o instanceof ListAssignMove)) {
+        if (o == null || getClass() != o.getClass()) {
             return false;
         }
-        ListAssignMove<?> that = (ListAssignMove<?>) o;
-        return destinationIndex == that.destinationIndex && variableDescriptor.equals(that.variableDescriptor)
-                && planningValue.equals(that.planningValue) && destinationEntity.equals(that.destinationEntity);
+        ListAssignMove<?> other = (ListAssignMove<?>) o;
+        return destinationIndex == other.destinationIndex
+                && Objects.equals(variableDescriptor, other.variableDescriptor)
+                && Objects.equals(planningValue, other.planningValue)
+                && Objects.equals(destinationEntity, other.destinationEntity);
```

Comment:
This is correct, in the sense that it's the way we do it now, so this should do it like this too.

But in general, we should open a ticket to wonder if our Move impls should rely on domain equality. That's error-prone and we have new philosophy since the Move impl were written to not depend on them. Any such pitfalls would only be uncovered by using MOVE_TABU, which almost no one does.

Reply:
https://issues.redhat.com/browse/PLANNER-2613